modules/management-console-agent/src/main/java/org/apache/ignite/agent/ws/WebSocketManager.java

```diff
@@ -259,136 +246,105 @@ private SslContextFactory createServerSslFactory(IgniteLogger log, ManagementCon
-            sslCtxFactory.setTrustAll(true);
-        }
-        else if (!F.isEmpty(trustStore)) {
-            try {
-                sslCtxFactory.setKeyStore(keyStore(trustStore, trustStorePwd));
-            }
-            catch (Exception e) {
-                log.warning("Failed to load server keyStore", e);
-            }
-        }
+        String proxyPort = System.getProperty("http.proxyPort");
 
-        if (!F.isEmpty(ciphers))
-            sslCtxFactory.setIncludeCipherSuites(ciphers.toArray(EMPTY));
+        if (!F.isEmpty(proxyHost) && !F.isEmpty(proxyPort)) {
+            mgr.getProperties().put(PROXY_URI, "http://" + proxyHost + ':' + proxyPort);
 
-        return sslCtxFactory;
+            addAuthentication(mgr);
+        }
     }
 
     /**
-     * @param srvUri Server uri.
+     * @param mgr Manager.
      */
-    private void configureProxy(IgniteLogger log, HttpClient httpClient, URI srvUri) {
-        try {
-            boolean secure = "https".equalsIgnoreCase(srvUri.getScheme());
-
-            List<ProxyConfiguration.Proxy> proxies = ProxySelector.getDefault().select(srvUri).stream()
-                .filter(p -> !p.equals(NO_PROXY))
-                .map(p -> {
-                    InetSocketAddress inetAddr = (InetSocketAddress)p.address();
+    private void addAuthentication(ClientManager mgr) {
+        String user = System.getProperty("http.proxyUsername");
 
-                    Origin.Address addr = new Origin.Address(inetAddr.getHostName(), inetAddr.getPort());
+        String pwd = System.getProperty("http.proxyPassword");
 
-                    if (p.type() == SOCKS)
-                        return new Socks4Proxy(addr, secure);
+        if (!F.isEmpty(user) || !F.isEmpty(pwd)) {
+            Map<String, String> proxyHeaders = new HashMap<>();
 
-                    return new HttpProxy(addr, secure);
-                })
-                .collect(toList());
+            proxyHeaders.put("Proxy-Authorization", "Basic " +
+                encodeToString((user + ':' + pwd).getBytes(Charset.forName("UTF-8"))));
 
-            httpClient.getProxyConfiguration().getProxies().addAll(proxies);
-
-            addAuthentication(httpClient, proxies);
-        }
-        catch (Exception e) {
-            log.warning("Failed to configure proxy.", e);
+            mgr.getProperties().put(ClientProperties.PROXY_HEADERS, proxyHeaders);
         }
     }
 
     /**
-     * @param httpClient Http client.
-     * @param proxies Proxies.
+     * Disabled trust manager, will skip all certificate checks.
      */
-    private void addAuthentication(HttpClient httpClient, List<ProxyConfiguration.Proxy> proxies) {
-        proxies.forEach(p -> {
-            String user, pwd;
-
-            if (p instanceof HttpProxy) {
-                String scheme = p.getURI().getScheme();
-
-                user = System.getProperty(scheme + ".proxyUsername");
+    private static class DisabledX509TrustManager implements X509TrustManager {
```

Comment:
Can be replaced by org.apache.ignite.ssl.SslContextFactory#getDisabledTrustManager

Reply:
Thanks :)